File: app/src/main/java/zone/com/zrefreshlayoutdemo/ScrollerViewActivity.java
package zone.com.zrefreshlayoutdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import butterknife.Bind;
import butterknife.ButterKnife;
import zone.com.zrefreshlayout.ZRefreshLayout;

/**
 * Created by fuzhipeng on 2017/1/10.
 */

public class ScrollerViewActivity extends AppCompatActivity {

    @Bind(R.id.tv1)
    TextView tv1;
    @Bind(R.id.tv2)
    TextView tv2;
    @Bind(R.id.tv3)
    TextView tv3;
    @Bind(R.id.refresh)
    ZRefreshLayout refresh;

    int i = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.scroller);
        ButterKnife.bind(this);

        initText();
        refresh.setLoadMoreListener(new ZRefreshLayout.LoadMoreListener() {
            @Override
            public void loadMore(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        tv1.setText("TV1:加载次数" + i);
                        tv2.setText("TV2:加载次数" + i);
                        tv3.setText("TV3:加载次数" + i);
                        zRefreshLayout.loadMoreComplete();
                    }
                }, 2000);

            }

        });
        refresh.setPullListener(new ZRefreshLayout.PullListener() {
            @Override
            public void refresh(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        i++;
                        initText();
                        zRefreshLayout.refreshComplete();
                    }
                }, 2000);
            }
        });
    }

    private void initText() {
        tv1.setText("TV1:刷新次数" + i);
        tv2.setText("TV2:刷新次数" + i);
        tv3.setText("TV3:刷新次数" + i);
    }
}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/ScrollerViewActivity.kt
package zone.com.zrefreshlayoutdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.scroller.*
import zone.com.zrefreshlayout.ZRefreshLayout

/**
 * Created by fuzhipeng on 2017/1/10.
 */

class ScrollerViewActivity : AppCompatActivity() {
    internal var i = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.scroller)

        initText()
        refresh.loadMoreListener = ZRefreshLayout.LoadMoreListener { zRefreshLayout ->
            refresh.postDelayed({
                tv1.text = "TV1:加载次数$i"
                tv2.text = "TV2:加载次数$i"
                tv3.text = "TV3:加载次数$i"
                zRefreshLayout.loadMoreComplete()
            }, 2000)
        }
        refresh.pullListener = ZRefreshLayout.PullListener { zRefreshLayout ->
            refresh.postDelayed({
                i++
                initText()
                zRefreshLayout.refreshComplete()
            }, 2000)
        }
    }

    private fun initText() {
        tv1.text = "TV1:刷新次数$i"
        tv2.text = "TV2:刷新次数$i"
        tv3.text = "TV3:刷新次数$i"
    }
}
